crates/storage/db/src/implementation/mdbx/cursor.rs

```diff
@@ -47,8 +51,8 @@ impl<K: TransactionKind, T: Table> Cursor<K, T> {
         value_size: Option<usize>,
         f: impl FnOnce(&mut Self) -> R,
     ) -> R {
-        if self.with_metrics {
-            OperationMetrics::record(T::NAME, operation, value_size, || f(self))
+        if let Some(metrics) = self.metrics.as_ref().cloned() {
```

Comment:
ok, we need this `cloned()` here because we pass `&mut self` to the closure below